test/test_gpu3.cpp

```diff
@@ -8738,6 +8738,82 @@ TEST_F(NVFuserTest, AvoidCachingSliceInput) {
     }
   }
 }
+
+// Step-1 to fix https://github.com/NVIDIA/Fuser/issues/1631
+// Needs to check consumer mapped with reduction inputs in
+// isReductionOutputMapped(), otherwise compute at root domain map
+// is wrong then leads to wrong compute at position and finally
+// expr sort failed.
+// After fix, there are two persistent buffers and can be further
+// reduced to one with a following step-2 to fix the issue in resolution
+// points detection.
+TEST_F(NVFuserTest, CaRootDomainMapConsumerMappedWithReductionInput) {
+  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
+  auto fusion = fusion_ptr.get();
+  FusionGuard fg(fusion);
+
+  const int batch_size = 128;
+  const int hidden_size = 10240;
+  DataType input_dtype = DataType::Half;
+  auto tv0 = makeContigTensor(2, input_dtype);
+  auto tv1 = makeContigTensor(2, DataType::Float);
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  auto tv2 = castOp(DataType::Float, tv0);
+  auto tv3 = sum(tv2, {1});
+  auto tv4 = broadcast(tv3, {false, true});
+  // manual project tv2 to input, so tv7 is mapped with tv2.
+  auto tv5 = castOp(DataType::Float, tv0);
+  auto tv6 = div(tv5, tv4);
+  auto tv7 = add(tv1, tv6);
+  auto tv8 = add(tv2, tv1);
+  fusion->addOutput(tv7);
+  fusion->addOutput(tv8);
+
+  // |--5------------------|
+  // 0 -> 2 -> r3 -> b4 -> 6 -> 7    8
+  // 1--------------------------|
+  // |-------------------------------|
+  //      |--------------------------|
+  // tv1 has two consumers, tv7 and tv8.
+  // tv7 is a consumer of the reduction output.
+  // tv8 is not a reduction input but it is mapped with a reduction input, tv2.
```

Comment:
Shouldn't this be:

```
If tv8 is mapped with `tv2`, we can't map tv7 and tv8...
```

Reply:
revised to:
```

  // |--5------------------|
  // 0 -> 2 -> r3 -> b4 -> 6 -> 8    9
  // 1-------> 7----------------|
  //           |---------------------|
  //      |--------------------------|
  // tv7 has two consumers, tv8 and tv9.
  // tv8 is a consumer of the reduction output.
  // if tv9 is mapped with tv2, we can't map tv8 and tv9.
  // Because tv9 is in the pre-reduction set and tv8 is in the post-reduction set.
```